aps2scala/dump-scala.cc

```diff
@@ -1447,6 +1448,85 @@ void dump_scala_Declaration_header(Declaration decl, ostream& oss)
   }
 }
 
+bool check_override_decl(Declaration decl, Declaration fdecl, std::set<Declaration> visited) {
+  if (visited.find(decl) != visited.end()) {
+    return false;
+  }
+
+  visited.insert(decl);
+
+  switch (Declaration_KEY(decl)) {
+  case KEYphylum_decl:
+    return check_override_decl(module_PHYLUM, fdecl, visited);
+  case KEYtype_decl:
+    return check_override_decl(module_TYPE, fdecl, visited);
+  case KEYsome_class_decl: {
+    Declaration result = some_class_decl_result_type(decl);
+    switch (Declaration_KEY(result)) {
+    case KEYsome_type_decl: {
+      Type rtype = some_type_decl_type(result);
+      if (Type_KEY(rtype) == KEYno_type) {
+        return check_override_decl(result, fdecl, visited);
+      }
+      break;
+    }
+    default:
+      break;
+    }
+
+    Block body = some_class_decl_contents(decl);
+    Declaration item;
+    int i = 0;
+    for (item = first_Declaration(block_body(body)); item != NULL; item = DECL_NEXT(item)) {
+      switch (Declaration_KEY(item)) {
+      case KEYsome_function_decl:
+        if (!strcmp(
+            symbol_name(def_name(some_function_decl_def(item))),
+            symbol_name(def_name(some_function_decl_def(fdecl))))) {
+
+          return true;
+        }
+        break;
+      default:
+        break;
+      }
+    }
+
+    return check_override_decl(result, fdecl, visited);
+  }
```

Comment:
Not quite.  This should check the `rtype`, or you might miss something.  Let me see if I can create a counter-example, before you change anything.  If my guess is right (and it might not be), then the example will miss an override.